src/main/scala/com/campudus/tableaux/database/model/structure/ColumnModel.scala

```diff
@@ -1,35 +1,65 @@
+  
+  def createColumn(table: Table, createColumn: CreateColumn): Future[ColumnType[_]] = {
+    createColumn match {
+      case CreateSimpleColumn(name, ordering, kind, languageType) =>
+        createValueColumn(table.id,kind, name, ordering, languageType).map {
+          case (id, ordering) => Mapper(languageType, kind).apply(table, id, name, ordering)
+        }
+
+      case CreateLinkColumn(name, ordering, linkConnection) => for {
+        toCol <- retrieve(linkConnection.toTableId, linkConnection.toColumnId).asInstanceOf[Future[SimpleValueColumn[_]]] 
+        (id, ordering) <- createLinkColumn(table.id, name, linkConnection, ordering)
+      } yield LinkColumn(table, id, toCol, name, ordering)
+
+      case CreateAttachmentColumn(name, ordering) =>
+        createAttachmentColumn(table.id, name, ordering).map {
+          case (id, ordering) => AttachmentColumn(table, id, name, ordering)
+        }
+    }
+  }
+  
+  def createValueColumn(tableId: TableId, dbType: TableauxDbType, name: String, ordering: Option[Ordering], languageType: LanguageType): Future[(ColumnId, Ordering)] = {
     connection.transactional { t =>
       for {
-        (t, result) <- insertSystemColumn(t, tableId, name, dbType, ordering, None)
+        (t, result) <- insertSystemColumn(t, tableId, name, dbType, ordering, None, languageType)
         result <- Future.successful(insertNotNull(result).head)
 
-        (t, _) <- t.query(s"ALTER TABLE user_table_$tableId ADD column_${result.get[ColumnId](0)} $dbType")
+        (t, _) <- languageType match {
+          case MultiLanguage => t.query(s"ALTER TABLE user_table_lang_$tableId ADD column_${result.get[ColumnId](0)} $dbType")
+          case SingleLanguage => t.query(s"ALTER TABLE user_table_$tableId ADD column_${result.get[ColumnId](0)} $dbType")
+        }
       } yield (t, (result.get[ColumnId](0), result.get[Ordering](1)))
     }
   }
 
-  def insertAttachment(tableId: TableId, name: String, ordering: Option[Ordering]): Future[(ColumnId, Ordering)] = {
+  def createAttachmentColumn(tableId: TableId, name: String, ordering: Option[Ordering]): Future[(ColumnId, Ordering)] = {
     connection.transactional { t =>
       for {
-        (t, result) <- insertSystemColumn(t, tableId, name, AttachmentType, ordering, None)
+        (t, result) <- insertSystemColumn(t, tableId, name, AttachmentType, ordering, None, SingleLanguage)
         result <- Future.successful(insertNotNull(result).head)
       } yield (t, (result.get[ColumnId](0), result.get[Ordering](1)))
     }
   }
 
-  def insertLink(tableId: TableId, name: String, fromColumnId: ColumnId, toTableId: TableId, toColumnId: ColumnId, ordering: Option[Ordering]): Future[(ColumnId, Ordering)] = {
+  def createLinkColumn(tableId: TableId, name: String, link: LinkConnection, ordering: Option[Ordering]): Future[(ColumnId, Ordering)] = {
```

Comment:
Would be nicer to return a `Future[<some case class>]` instead of `Future[some tuple]`